scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1727,3 +1727,178 @@ def test_len_and_bool():
+            angles[:, 1] = angles_middle - lamb
+            rot = Rotation.from_davenport(ax, 'extrinsic', angles)
+            mat_expected = rot.as_matrix()
+            angles_dav = rot.as_davenport(ax, 'extrinsic')
+            mat_estimated = Rotation.from_davenport(
+                ax, 'extrinsic', angles_dav).as_matrix()
+            assert_array_almost_equal(mat_expected, mat_estimated)
+
+
+def test_compare_from_davenport_from_euler():
+
+    def basis_vec(axis):
+        if axis == 'x':
+            return [1, 0, 0]
+        elif axis == 'y':
+            return [0, 1, 0]
+        elif axis == 'z':
+            return [0, 0, 1]
+
+    rnd = np.random.RandomState(0)
+    n = 100
+    angles = np.empty((n, 3))
+
+    # symmetric sequences
+    angles[:, 0] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    angles[:, 1] = rnd.uniform(low=0, high=np.pi, size=(n,))
+    angles[:, 2] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    for order in ['extrinsic', 'intrinsic']:
+        for seq_tuple in permutations('xyz'):
+            seq = ''.join([seq_tuple[0], seq_tuple[1], seq_tuple[0]])
+            ax = [basis_vec(i) for i in seq]
+            if order == 'intrinsic':
+                seq = seq.upper()
+            eul = Rotation.from_euler(seq, angles)
+            dav = Rotation.from_davenport(ax, order, angles)
+            assert_allclose(eul.as_quat(), dav.as_quat(), atol=0, rtol=1e-9)
+
+    # asymmetric sequences
+    angles[:, 1] -= np.pi / 2
+    for order in ['extrinsic', 'intrinsic']:
+        for seq_tuple in permutations('xyz'):
+            seq = ''.join(seq_tuple)
+            ax = [basis_vec(i) for i in seq]
+            if order == 'intrinsic':
+                seq = seq.upper()
+            eul = Rotation.from_euler(seq, angles)
+            dav = Rotation.from_davenport(ax, order, angles)
+            assert_allclose(eul.as_quat(), dav.as_quat(), atol=0, rtol=1e-9)
+
+
+def test_compare_as_davenport_as_euler():
+
+    def basis_vec(axis):
+        if axis == 'x':
+            return [1, 0, 0]
+        elif axis == 'y':
+            return [0, 1, 0]
+        elif axis == 'z':
+            return [0, 0, 1]
+
+    rnd = np.random.RandomState(0)
+    n = 100
+    angles = np.empty((n, 3))
+
+    # symmetric sequences
+    angles[:, 0] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    angles[:, 1] = rnd.uniform(low=0, high=np.pi, size=(n,))
+    angles[:, 2] = rnd.uniform(low=-np.pi, high=np.pi, size=(n,))
+    for order in ['extrinsic', 'intrinsic']:
+        for seq_tuple in permutations('xyz'):
+            seq = ''.join([seq_tuple[0], seq_tuple[1], seq_tuple[0]])
+            ax = [basis_vec(i) for i in seq]
+            if order == 'intrinsic':
+                seq = seq.upper()
+            rot = Rotation.from_euler(seq, angles)
+            eul = rot.as_euler(seq)
+            dav = rot.as_davenport(ax, order)
+            assert_allclose(eul, dav, atol=0, rtol=1e-9)
```

Comment:
I guess `atol` might be technically necessary, because there is no guarantee that generated angles are not very close to zero. Not sure what is the appropriate value.